beacon-chain/sync/backfill/backfill.md

```diff
@@ -0,0 +1,853 @@
+- Determines which batches are importable (ready for DB import)
+- Batches must be imported in descending order to maintain chain continuity
+
+**Location**: `beacon-chain/sync/backfill/batcher.go`
+
+#### 4. Worker Pool (`pool.go`)
+Concurrent download system:
+- Spawns N workers (configurable, default 2)
+- **Batch Router**: Assigns available peers to pending batches
+- Manages peer busy state to avoid overloading peers
+- Routes completed batches back to service
+
+**Location**: `beacon-chain/sync/backfill/pool.go`
+
+#### 5. Workers (`worker.go`)
+Individual download workers:
+- Download blocks via `BeaconBlocksByRange` RPC
+- Download blobs via `BlobSidecarsByRange` RPC (if needed)
+- Verify blocks using the verifier
+- Handle both block and blob sync phases
+
+**Location**: `beacon-chain/sync/backfill/worker.go`
+
+#### 6. Verifier (`verify.go`)
+Block verification:
+- Verifies proposer signatures using BLS batch verification
+- Checks parent-child chain continuity (parent_root matches)
+- Uses signing domains cached by epoch for performance
+- Validator public keys loaded from checkpoint state
+
+**Location**: `beacon-chain/sync/backfill/verify.go`
+
+#### 7. Blob Sync (`blobs.go`)
+Handles blob sidecars for Deneb+ blocks:
+- Identifies which blocks need blobs (post-Deneb with commitments)
+- Downloads blob sidecars separately
+- Verifies KZG commitments and proofs
+- Stores blobs to filesystem storage
+
+**Location**: `beacon-chain/sync/backfill/blobs.go`
+
+## How It Works
+
+### Initialization Phase
+
+1. **Service Creation** (`NewService`)
+   - Store initialized with backfill status from DB (or recovered from legacy checkpoint sync)
+   - Worker pool created with configured worker count
+   - Verifier prepared (waits for genesis data)
+
+2. **Service Start** (`Start`)
+   ```
+   ┌─────────────────────────────────────────┐
+   │ Check if enabled (--enable-backfill)    │
+   └────────────┬────────────────────────────┘
+                ↓
+   ┌─────────────────────────────────────────┐
+   │ Wait for clock (genesis data)           │
+   └────────────┬────────────────────────────┘
+                ↓
+   ┌─────────────────────────────────────────┐
+   │ Check if genesis sync (short-circuit)   │
+   └────────────┬────────────────────────────┘
+                ↓
+   ┌─────────────────────────────────────────┐
+   │ Check if already complete               │
+   │ (LowSlot <= minimumBackfillSlot)        │
+   └────────────┬────────────────────────────┘
+                ↓
+   ┌─────────────────────────────────────────┐
+   │ Initialize verifier with origin state   │
+   └────────────┬────────────────────────────┘
+                ↓
+   ┌─────────────────────────────────────────┐
+   │ Wait for initial sync to complete       │
+   └────────────┬────────────────────────────┘
+                ↓
+   ┌─────────────────────────────────────────┐
+   │ Spawn workers and start main loop       │
+   └─────────────────────────────────────────┘
+   ```
```

Comment:
Why a diagram for this? I feel nitpicky commenting on this but it seems like overkill for this to be a diagram when there is no branching, it's just describing a sequence of actions that need to succeed.